src/main/java/emissary/config/ServiceConfigGuide.java

```diff
@@ -323,7 +324,10 @@ protected ConfigEntry handleNewEntry(final String parmNameArg, final String sval
                 } catch (IOException e) {
                     // Throw exception if it is an IMPORT_FILE and the base file is not found
                     if ("IMPORT_FILE".equals(parmName) && i == 0) {
-                        throw new IOException("IMPORT_FILE = " + sval + " : Directive failed. Called from " + filename, e);
+                        String importFileName = Paths.get(svalArg).getFileName().toString();
+                        throw new IOException("In " + filename + ", cannot find IMPORT_FILE: " + sval
```

Comment:
Out of caution and without testing to be sure, I would shy away from using \n in any logs or exceptions due to the possibility of breaking any expected logstash grok-ing.

Reply:
Can quickly go in and remove it. Had it originally for readability ease.

Reply:
I don't think this is fixing the issue. In this class we are still throwing the IOException, but with a different message. The ticket say something about how we are not handling the exception correctly. I updated the JsonEscapePlace.cfg file and added IMPORT_FILE = "/does/not/exist" and got the following error:

> 2021-12-28 22:58:56,000 localhost-8001 ERROR emissary.core.Factory -   - Constructor failed
java.lang.reflect.InvocationTargetException: null
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at emissary.core.Factory.create(Factory.java:67)
        at emissary.core.Factory.create(Factory.java:122)
        at emissary.admin.PlaceStarter.createPlace(PlaceStarter.java:107)
        at emissary.admin.PlaceStarter.createPlace(PlaceStarter.java:72)
        at emissary.admin.Startup$1.run(Startup.java:371)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: None of the 2 preferences could be found: [emissary.transform.JsonEscapePlace.cfg, JsonEscapePlace.cfg]
        at emissary.config.ConfigUtil.getConfigInfo(ConfigUtil.java:333)
        at emissary.place.ServiceProviderPlace.loadConfigurator(ServiceProviderPlace.java:215)
        at emissary.place.ServiceProviderPlace.loadConfigurator(ServiceProviderPlace.java:193)
        at emissary.place.ServiceProviderPlace.<init>(ServiceProviderPlace.java:139)
        at emissary.transform.JsonEscapePlace.<init>(JsonEscapePlace.java:30)
        ... 10 common frames omitted

The import failed, but it looks like the JsonEscapePlace.cfg does not exist.